discoverx/explorer.py

```diff
@@ -0,0 +1,287 @@
+        else:
+            raise ValueError(
+                f"Invalid from_tables statement '{from_tables}'. Should be a string in format 'table_catalog.table_schema.table_name'. You can use '*' as wildcard."
+            )
+
+    def __deepcopy__(self, memo):
+        new_obj = type(self).__new__(self.__class__)
+        new_obj.__dict__.update(self.__dict__)
+        new_obj._catalogs = copy.deepcopy(self._catalogs)
+        new_obj._schemas = copy.deepcopy(self._schemas)
+        new_obj._tables = copy.deepcopy(self._tables)
+        new_obj._having_columns = copy.deepcopy(self._having_columns)
+        new_obj._sql_query_template = copy.deepcopy(self._sql_query_template)
+        new_obj._max_concurrency = copy.deepcopy(self._max_concurrency)
+
+        new_obj._spark = self._spark
+        new_obj._info_fetcher = self._info_fetcher
+
+        return new_obj
+
+    def having_columns(self, *columns) -> "DataExplorer":
+        new_obj = copy.deepcopy(self)
+        new_obj._having_columns.extend(columns)
+        return new_obj
+
+    def with_concurrency(self, max_concurrency) -> "DataExplorer":
+        new_obj = copy.deepcopy(self)
+        new_obj._max_concurrency = max_concurrency
+        return new_obj
+
+    def with_sql(self, sql_query_template: str) -> "DataExplorerActions":
+        new_obj = copy.deepcopy(self)
+        new_obj._sql_query_template = sql_query_template
+        return DataExplorerActions(new_obj, spark=self._spark, info_fetcher=self._info_fetcher)
+
+    def melt_string_columns(self, sample_size=None) -> "DataExplorerActions":
+        sql_query_template = """
+        SELECT
+            {stack_string_columns} AS (column_name, string_value)
+        FROM {full_table_name}
+        """
+        if sample_size is not None:
+            sql_query_template += f"TABLESAMPLE ({sample_size} ROWS)"
+
+        return self.with_sql(sql_query_template)
+
+
+class DataExplorerActions:
+    def __init__(
+        self, data_explorer: DataExplorer, spark: SparkSession = None, info_fetcher: InfoFetcher = None
+    ) -> None:
+        self._data_explorer = data_explorer
+        if spark is None:
+            spark = SparkSession.builder.getOrCreate()
+        if info_fetcher is None:
+            info_fetcher = InfoFetcher(spark)
+        self._info_fetcher = info_fetcher
+        self._spark = spark
+
+    @staticmethod
+    def _get_stack_string_columns_expression(table_info: TableInfo) -> str:
+        string_col_names = [c.name for c in table_info.columns if c.data_type.lower() == "string"]
+        stack_parameters = ", ".join([f"'{c}', `{c}`" for c in string_col_names])
+        return f"stack({len(string_col_names)}, {stack_parameters})"
+
+    @staticmethod
+    def _build_sql(sql_template: str, table_info: TableInfo) -> str:
+        if table_info.catalog and table_info.catalog != "None":
+            full_table_name = f"{table_info.catalog}.{table_info.schema}.{table_info.table}"
+        else:
```

Comment:
I would prefer to keep this here, so that we can make non-UC integration easier when/if we go for it.
Mocking it would reduce the coverage metrics ;P